src/ConnectionPool.jl

```diff
@@ -451,14 +451,24 @@ function getconnection(::Type{Transaction{T}},
     end
 end
 
+function keepalive!(tcp)
+    @debug 2 "setting keepalive on tcp socket"
+    err = ccall(:uv_tcp_keepalive, Cint, (Ptr{Cvoid}, Cint, Cuint),
+                                          tcp.handle, 1, 1)
```

Comment:
@samoconnor @Keno, the 2nd argument here is an "initial delay", which I assume is how long it waits before sending the first keepalive packet (it's unclear how frequently it sends it after that, but whatever). I just set this to 1 for now, but if you guys have other opinions or concerns, I'm happy to try whatever here.

Reply:
Every time I've looked into the libuv docs I find it very hard to figure out what the exact behaviour is supposed to be. They often just say "see the linux man page for more" without explaining how their behaviour differs. (Sometimes I wonder if we'd be better off using BSD sockets directly and using some kind of BSD/Winsock layer to make it work on windows. libuv is a weird beast).

I think in this case it would make sense to observe what goes on with wireshark.
Probably OSX, Linux and Windows will behave differently too.

Perhaps you can find some place in libcurl, or in python's requests library where they set keepalive options and see if there are any clues there...